### ISSUE TITLE
Player entity stops detecting collisions when elasticity is set to below 1

### ISSUE THREAD

**Issue by unchangingConstant** [unchangingConstant/2DPhysicsEngine#1]
For some reason, whenever the elasticity is set to below 1 but above 0, and the entity bounces an even number of times, there is a chance it'll simply ignore a surface. Here's a video demonstrating the issue:

https://github.com/unchangingConstant/2DPhysicsEngine/assets/76669746/6d0d5a6d-59b6-4230-a8cc-5ed1353da3f4

**Comment by unchangingConstant** [comment]
Another helpful bit: I recorded the entity's velocity and position before the bug occurs. And it replicates the issue every time. Here's a vid:

https://github.com/unchangingConstant/2DPhysicsEngine/assets/76669746/2ab7ae21-d961-4959-93fc-4904797c65fa

**Comment by unchangingConstant** [comment]
Found the issue. I added some print statements to the collisionDetected() function and got the following returns:

CODE:
        xPathBound = (entity.priorPosition[0], entity.position[0])
        yPathBound = (entity.priorPosition[1], entity.position[1])

        xSurfaceBound = (surface.edge1[0], surface.edge2[0])
        ySurfaceBound = (surface.edge1[1], surface.edge2[1])

        print(collisionPoint[1], end = ' ')
        print(entity.priorPosition[1], entity.position[1], end = ' ')
        print(yPathBound)
        entityPathIntersects = pointInBounds(collisionPoint, xPathBound, yPathBound)
        surfaceIntersects = pointInBounds(collisionPoint, xSurfaceBound, ySurfaceBound)
`

OUTPUT (At point where entity crosses the bound)
`[ 6.2644 200.    ] 200.00000000000003 200.0 200.0 (200.0, 200.0)
[  0.3595 200.    ] 200.00000000000003 200.0 200.0 (200.0, 200.0)
[ -5.5454 200.    ] 200.00000000000003 200.0 200.0 (200.0, 200.0)`

So... bruh. It's a rounding issue. 
I'm gonna try and root out any code that creates decimal -y numbers to see if that removes the issue forever. Likely will have to add some rounding to things.

**Comment by unchangingConstant** [comment]
I've decided to go about this by making an entity's velocity and position private fields, that can only be modified with methods that ensure the fields remain rounded to whole numbers.

**Comment by unchangingConstant** [comment]
I've implemented private entity fields for velocity, position, and priorPosition as well as corresponding setter, getter and mutator methods. The new methods make sure that the fields always remain rounded to the nearest whole number. This presents some new problems regarding elasticity that I will address in a separate issue though. For now though, collisions consistently work.

I also deleted some (for now) unused or unnecessary fields and methods like tick(), force(), walkSpeed and dex. 
The new changes will be pushed via the roundingNumbers branch
